griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,400 @@
+            )
+
+    @activity(
+        config={
+            "description": "Can be used to share a spreadsheet with a specified user.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="The path of the spreadsheet to share",
+                    ): str,
+                    Literal(
+                        "email_address",
+                        description="The email address of the user to share with",
+                    ): str,
+                    Optional(
+                        "role",
+                        default="reader",
+                        description="The role to give to the user, e.g., 'reader', 'writer', or 'commenter'",
+                    ): Or("reader", "writer", "commenter"),
+                }
+            ),
+        }
+    )
+    def share_spreadsheet(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        file_path = params["file_path"]
+        email_address = params["email_address"]
+        role = params.get("role", "reader")
+
+        try:
+            service = self._build_client(
+                scopes=self.DRIVE_AUTH_SCOPES,
+                service_name="drive",
+                version="v3",
+                owner_email=self.owner_email,
+            )
+
+            if file_path.lower() == self.DEFAULT_FOLDER_PATH:
+                spreadsheet_id = self.DEFAULT_FOLDER_PATH
+            else:
+                spreadsheet_id = self._convert_path_to_file_id(
+                    service, file_path
+                )
+
+            if spreadsheet_id:
+                batch_update_permission_request_body = {
+                    "role": role,
+                    "type": "user",
+                    "emailAddress": email_address,
+                }
+                request = service.permissions().create(
+                    fileId=spreadsheet_id,
+                    body=batch_update_permission_request_body,
+                    fields="id",
+                )
+                request.execute()
+                return InfoArtifact(
+                    f"Spreadsheet at {file_path} shared with {email_address} as a {role}"
+                )
+            else:
+                return ErrorArtifact(
+                    f"error finding spreadsheet at path: {file_path}"
+                )
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error sharing spreadsheet: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to check the permissions on a specified spreadsheet.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="The path of the spreadsheet to check permissions for",
+                    ): str,
+                }
+            ),
+        }
+    )
+    def check_permissions_for_spreadsheet(
```

Comment:
Should this be called `list_permissions_for_spreadsheet`? Checking, to me, implies some sort of definite answer. I.e. yes you have permission.

Reply:
Sounds good, i'll change the name!